e2e/imgbuild/regressions.go

```diff
@@ -393,3 +393,24 @@ func (c *imgBuildTests) issue5315(t *testing.T) {
 		),
 	)
 }
+
+// This test will attempt to build an image by passing an empty string as
+// the build destination. This should fail.
+func (c *imgBuildTests) issue5435(t *testing.T) {
+	// create a directory that we don't care about
+	cwd, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "throwaway-dir-", "")
+
+	c.env.RunSingularity(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("", ""),
+		e2e.WithDir(cwd),
+		e2e.PostRun(func(t *testing.T) {
+			if !t.Failed() {
+				cleanup(t)
+			}
+		}),
+		e2e.ExpectExit(255),
```

Comment:
I would suggest testing to see that the `cwd` still exists here, or confirm the error message relates to *not* removing the existing directory, and not just rely on the exit code.

It is possible that we could delete a directory, but fail with a 255 error later on for a different reason.

Reply:
Tested on master to make sure it's properly checking the CWD
```
=== RUN   TestE2E/PAR/BUILD/issue_5435
=== PAUSE TestE2E/PAR/BUILD/issue_5435
=== CONT  TestE2E/PAR/BUILD/issue_5435
    TestE2E/PAR/BUILD/issue_5435: singularitycmd.go:663: Running command "/usr/local/bin/singularity build  "
    TestE2E/PAR/BUILD/issue_5435: singularitycmd.go:663: 
        "/usr/local/bin/singularity build  " output:
        INFO:    Starting build...
        INFO:    Creating SIF file...
        INFO:    Architecture not recognized, use native
        INFO:    Build complete: 
        
    TestE2E/PAR/BUILD/issue_5435: singularitycmd.go:663: unexpected success while executing "/usr/local/bin/singularity build  "
    TestE2E/PAR/BUILD/issue_5435: regressions.go:426: cwd overwritten
```